File: public-api/src/main/java/com/shopleech/publicapi/dal/repository/OfferRepository.java
package com.shopleech.publicapi.dal.repository;

import com.shopleech.publicapi.domain.Offer;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.stereotype.Repository;

import java.util.List;

/**
 * @author Ahto Jalak
 * @since 29.03.2023
 */
@Repository
public interface OfferRepository extends JpaRepository<Offer, Integer>, OfferRepositoryCustom {
    @Query(value = "SELECT * FROM offer where product_id = :id", nativeQuery = true)
    List<Offer> getAllByProductId(Integer id);

    @Query(value = "SELECT * FROM offer where product_id = :productId AND account_id = :accountId " +
            "AND shop_id = :shopId", nativeQuery = true)
    Offer findByProduct(Integer accountId, Integer productId, Integer shopId);
}

File: price-comparison-client1/src/dal/repository/OfferRepository.ts
import { BaseRepository } from '@/dal/repository/BaseRepository'
import type { IOfferRepository } from '@/dal/repository/model/IOfferRepository'
import httpCLient from '@/util/http-client'
import { AxiosError } from 'axios'
import type { IOffer } from '@/dal/domain/IOffer'
import type { IServiceResult } from '@/bll/service/model/IServiceResult'

/**
 * @author Ahto Jalak
 * @since 12.04.2023
 */
export class OfferRepository extends BaseRepository<IOffer> implements IOfferRepository {
    constructor () {
        super('v1/offer')
    }

    async getOfferListById (productId: number): Promise<IServiceResult<IOffer[]>> {
        this.logger.info('findByName')

        let response
        try {
            response = await httpCLient.get(`/${this._path}/product/${productId}`, {
                headers: {
                    Authorization: 'Bearer ' + this.identityStore.$state.jwt?.token,
                }
            })
        } catch (e:any) {
            return {
                status: (e as AxiosError).response?.status,
                errorMsg: e.response?.data.message,
            }
        }

        return {
            status: response.status,
            data: response.data.details,
        }
    }
}
